### PR TITLE
Issue 256:  Enable stream tag API on the RUST client.

### LINKED ISSUE #256: Enable Stream tag APIs on the RUST client.
**Problem description**
Implement Stream tag APIs on the RUST client.

**Problem location**
Controller Client

**Suggestions for an improvement**
Implement it.

### PR DESCRIPTION
**Pull request by shrids**
**Change log description**  
- Enable Stream Tag API on the RUST client.

**Purpose of the change**  
Fixes #256 

**What the code does**  
- Update the Controller proto file with the latest version of the Controller.
- Enable Tag related APIs on the RUST client.
The following APIs have been enabled.

```rust

    /**
     * API to fetch the Stream Configuration of a Stream.
     */
    async fn get_stream_configuration(&self, stream: &ScopedStream) -> ResultRetry<StreamConfiguration>;

    /**
     * API to fetch the Tags for a Stream.
     */
    async fn get_stream_tags(&self, stream: &ScopedStream) -> ResultRetry<Option<Vec<String>>>;

    /**
     * API to list streams associated with the given tag under a given scope and continuation token.
     * Use the pravega_controller_client::paginator::list_streams_for_tag to paginate over all the streams.
     */
    async fn list_streams_for_tag(
        &self,
        scope: &Scope,
        tag: &str,
        token: &CToken,
    ) -> ResultRetry<Option<(Vec<ScopedStream>, CToken)>>;
```
- Enable CLI option to create Stream with tags and list streams with a tag under a given scope.
```
osboxes@osboxes:~$ ./controller-cli list-streams-for-tag --help
controller-cli-list-streams-for-tag 0.2.0
List Streams for a tag, under a scope

USAGE:
    controller-cli list-streams-for-tag <scope-name> <tag>

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information

ARGS:
    <scope-name>    Scope Name
    <tag>           Tag Name

```

**How to verify it**  
All the existing and newly added tests should pass.
Integration tests have been created to run against the latest version of Pravega.

### CODE REVIEW

File: controller-client/proto/Controller.proto [hunk 1]
```diff
@@ -1,11 +1,17 @@
 /**
- * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * Copyright Pravega Authors.
```
Comment (Tristan1900) [inline controller-client/proto/Controller.proto:2]: I guess we need to replace this everywhere?

Comment (shrids) [reply controller-client/proto/Controller.proto:2]: I have created an issue https://github.com/pravega/pravega-client-rust/issues/272 to fix this seperately.

**Comment by shrids** [comment]
@tkaitchuck @Tristan1900 The Pravega PR for Stream tags has been merged. Could you review this changeset ?

**Review by tkaitchuck — APPROVED** [review]

File: controller-client/src/paginator.rs [hunk 2]
```diff
@@ -129,3 +129,123 @@ pub fn list_streams(
...
     )
 }
+
+///
+///Helper method to iterated over the all the Pravega streams under the provided Scope.
+///This method returns a stream of values,Pravega streams, produced asynchronously.
+///
+/// The below snippets show case the example uses.
+/// Sample 1:
+///```ignore
```
[the remark this answers is not in the document]
Comment (shrids) [reply controller-client/src/paginator.rs]: thanks, I have improved the snippet to enable compilation check.

File: controller-client/src/paginator.rs [hunk 3]
```diff
@@ -129,3 +129,123 @@ pub fn list_streams(
...
+///         "tagx".to_string(),
+///         controller_client,
+///     );
+///     // collect all the Streams in a single vector
+///     let stream_list:Vec<ScopedStream> = stream.map(|str| str.unwrap()).collect::<Vec<ScopedStream>>().await;
+///  # });
+/// ```
+///
+/// Sample 2:
+/// ```ignore
```
[the remark this answers is not in the document]
Comment (shrids) [reply controller-client/src/paginator.rs]: fixed it.

**Review by Tristan1900 — APPROVED** [review]

**Pull request merged**